src/stores/WalletStore.ts

```diff
@@ -81,75 +120,123 @@ export default class WalletStore {
-      }, () => console.log('Account added', account));
-      this.accounts = accounts;
-      this.loggedInAccount = account;
+        [STORAGE.MAINNET_ACCOUNTS]: toJS(this.mainnetAccounts),
+      }, () => console.log('Mainnet Account added', account));
+    } else {
+      this.testnetAccounts.push(account);
+      chrome.storage.local.set({
+        [STORAGE.TESTNET_ACCOUNTS]: toJS(this.testnetAccounts),
+      }, () => console.log('Testnet Account added', account));
     }
+
+    this.loggedInAccount = account;
+    await this.onAccountLoggedIn();
   }
 
+  /*
+  * Finds the account based on the name and logs in.
+  * @param accountName {string} The account name to search by.
+  */
   @action
-  public async login(accountName: string) {
-    const foundAccount = find(this.accounts, { name: accountName });
+  public async loginAccount(accountName: string) {
+    this.loading = true;
+
+    const accounts = this.app.networkStore.isMainNet ? this.mainnetAccounts : this.testnetAccounts;
+    const foundAccount = find(accounts, { name: accountName });
     if (foundAccount) {
       this.loggedInAccount = foundAccount;
-      await this.recoverWallet(this.loggedInAccount!.mnemonic!);
-      await this.startPolling();
-      runInAction(() => {
-        this.loading = false;
-        this.app.routerStore.push('/home');
-      });
+
+      // Recover wallet
+      const network = this.app.networkStore.network;
+      this.wallet = await network.fromEncryptedPrivateKey(this.loggedInAccount!.privateKeyHash, this.passwordHash);
+
+      await this.onAccountLoggedIn();
     }
   }
 
   @action
-  public logout = (isSwitchingNetwork: boolean) => {
+  public logout = () => {
     this.stopPolling();
     this.info =  INIT_VALUES.info;
     this.loggedInAccount = INIT_VALUES.loggedInAccount;
     this.wallet = INIT_VALUES.wallet;
-
-    if (isSwitchingNetwork) {
-      this.accounts = INIT_VALUES.accounts;
-      this.app.walletStore.getAccountsFromStorage();
-      // we dont call this.app.routerStore.push('/login') here because it is called at the end of getAccountsFromStorage() instead
-    } else {
-      this.app.routerStore.push('/login');
-    }
+    this.routeToAccountPage();
   }
 
-  @action
-  private recoverWallet = async (mnemonic: string) => {
-    const network = this.app.networkStore.network;
-    this.wallet = await network.fromMnemonic(mnemonic);
+  /*
+  * Initializes all the values from Chrome storage on startup.
+  */
+  private fetchStorageValues = () => {
+    const { APP_SALT, PASSWORD_HASH, MAINNET_ACCOUNTS, TESTNET_ACCOUNTS } = STORAGE;
+    chrome.storage.local.get([APP_SALT, PASSWORD_HASH, MAINNET_ACCOUNTS, TESTNET_ACCOUNTS],
+      ({ appSalt, passwordHash, mainnetAccounts, testnetAccounts }: any) => {
+        if (!isEmpty(appSalt)) {
+          const array = split(appSalt, ',').map((str) => parseInt(str, 10));
+          this.appSalt =  Uint8Array.from(array);
+        }
+
+        if (!isEmpty(passwordHash)) {
```

Comment:
hmmm, i think we shouldn't store passwordHash in storage? It should only be kept in memory.

Reply:
as discussed, will remove this and change the flow for the first time user

Reply:
I ended up keeping the `Login` page as the initial page with a note that the master pw wont save unless they create/import their first account.